var/spack/repos/builtin/packages/geant4/package.py

```diff
@@ -316,6 +320,9 @@ def cmake_args(self):
                 options.append(self.define("GEANT4_USE_QT_QT6", True))
             options.append(self.define("QT_QMAKE_EXECUTABLE", spec["qmake"].prefix.bin.qmake))
 
+        if "+hdf" in spec:
+            options.append(self.define("GEANT4_USE_HDF5", True))
```

Comment:
This should work and be more foolproof (explicitly setting it to "off" if disabled)
```suggestion
        options.append(self.define_from_variant("GEANT4_USE_HDF5", "hdf"))
```